src/js/components/search/filters/agency/AgencyList.jsx

```diff
@@ -0,0 +1,219 @@
+import SelectedAgencies from './SelectedAgencies';
+
+const propTypes = {
+    autocompleteAwardingAgencies: PropTypes.array,
+    autocompleteFundingAgencies: PropTypes.array,
+    onSelect: PropTypes.func,
+    customClass: PropTypes.string,
+    keyValue: PropTypes.string,
+    internalValue: PropTypes.string,
+    formatter: React.PropTypes.func,
+    tabIndex: PropTypes.number,
+    handleTextInput: PropTypes.func,
+    isRequired: PropTypes.bool,
+    agencyType: PropTypes.string,
+    checkValidity: PropTypes.func
+};
+
+const defaultProps = {
+    customClass: '',
+    formatter: null,
+    keyValue: 'agency',
+    internalValue: 'agency_name',
+    tabIndex: null,
+    isRequired: false,
+    errorHeader: '',
+    errorMessage: ''
+};
+
+export default class AgencyList extends Typeahead {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            showWarning: false,
+            errorMessage: null,
+            errorHeader: null
+        };
+
+        this.timeout = null;
+    }
+
+    loadValues() {
+        const awardingValuesList = [];
+        const fundingValuesList = [];
+        if (this.props.autocompleteAwardingAgencies.length > 0) {
+            this.props.autocompleteAwardingAgencies.forEach((item) => {
+                awardingValuesList.push(item);
+                const key =
+                `<b>${item}</b>`;
+                this.dataDictionary[key] = item;
+            });
+        }
+        if (this.props.autocompleteFundingAgencies.length > 0) {
+            this.props.autocompleteFundingAgencies.forEach((item) => {
+                fundingValuesList.push(item);
+                const key =
+                `<b>${item}</b>`;
+                this.dataDictionary[key] = item;
+            });
+        }
+
+        if (this.props.agencyType === "Awarding") {
+            this.typeahead.list = awardingValuesList;
+        }
+        else {
+            this.typeahead.list = fundingValuesList;
+        }
+
+        this.typeahead.replace = () => {
+            this.typeahead.input.value = "";
+        };
+    }
+
+    componentDidUpdate(prevProps) {
+        if (!_.isEqual(prevProps.autocompleteAwardingAgencies,
+            this.props.autocompleteAwardingAgencies) ||
+            !_.isEqual(prevProps.autocompleteFundingAgencies,
+                this.props.autocompleteFundingAgencies)) {
+            this.loadValues();
+        }
+    }
+
+    bubbleUpChange() {
+        // Force the change up into the parent components
+        // Validate the current value is on the autocomplete list
+        let selectedAgency = null;
+        let isValid = false;
+        const key = this.dataDictionary[`<b>${this.state.value}</b>`];
+        if (key !== null) {
+            isValid = true;
+        }
+        if (isValid) {
+            // Find matching agency object from redux store
+            if (this.props.agencyType === "Awarding") {
```

Comment:
We can probably streamline this logic by using programmatically modified variable key strings. So:

```
const autocompleteKey = `autocomplete${this.props.agencyType}Agencies`;
....
this.props[autocompleteKey]
```